flexget/plugins/input/my_anime_list.py

```diff
@@ -0,0 +1,119 @@
+            'username': {'type': 'string'},
+            'status': one_or_more({'type': 'string', 'enum': ['watching', 'completed', 'on_hold', 'dropped', 'plan_to_watch']}),
+            'type': one_or_more({'type': 'string', 'enum': ['series', 'ova', 'movie', 'ona']})
+        },
+        'required': ['username'],
+        'additionalProperties': False
+    }
+
+    @cached('my_anime_list', persist='2 hours')
+    def on_task_input(self, task, config):
+        entries = []
+        parameters = {'u': config['username'], 'status': 'all', 'type': 'anime'}
+
+        if config.has_key('status'):
+            selected_status = config['status']
+        else:
+            selected_status = STATUS.values()
+
+        if config.has_key('type'):
+            selected_types = config['type']
+        else:
+            selected_types = ANIME_TYPE.values()
+
+        try:
+            list_response = task.requests.get('https://myanimelist.net/malappinfo.php', params=parameters)
+            tree = ET.fromstring(list_response.text.encode('utf-8'))
+            list_items = tree.findall('anime')
+        except RequestException as e:
+            raise plugin.PluginError('Error finding list on url: {url}'.format(url=e.request.url))
+        except ET.ParseError:
+            raise plugin.PluginError('Bad XML')
+
+        for item in list_items:
+            my_anime_list_id = item.findtext('series_animedb_id')
+            title = item.findtext('series_title').strip()
+            anime_type = ANIME_TYPE[item.findtext('series_type', 1)]
+            my_status = STATUS[item.findtext('my_status')]
+
+            my_tags = []
+            alternate_names = []
+            isExact = False
+
+            for name in item.findtext('series_synonyms', '').split('; '):
+                stripped = name.strip()
+                if stripped and stripped is not title:
+                    alternate_names.append(name)
+
+            for tag in item.findtext('my_tags', '').split(','):
+                stripped = tag.strip()
+                if stripped:
+                    my_tags.append(stripped)
+                    if stripped is 'exact':
+                        isExact = True
+
+            if my_status in selected_status and anime_type in selected_types:
+                entry = Entry(title=title,
+                              url='https://myanimelist.net/anime/{}'.format(my_anime_list_id),
+                              configure_series_alternate_name=alternate_names,
+                              my_anime_list_type=anime_type,
+                              my_anime_list_status= my_status,
+                              my_anime_list_tags=my_tags)
+
+                if isExact:
+                    entry['configure_series_exact'] = True
+
+                if entry.isvalid():
+                    entries.append(entry)
+
+        return entries
+
+
+@event('plugin.register')
+def register_plugin():
+    plugin.register(MyAnimeList, 'my_anime_list', api_ver=2)
```

Comment:
I gratefully stole general structure from `kitsu.py` since this is my first plugin. That one didn't have any interfaces in the statement, so I wasn't exactly sure about what to do with them, since I was not able to find any docs for this.